src/RequestHandler.cpp

```diff
@@ -86,13 +85,22 @@ void RequestHandler::showTopicPartitionOffsets(
   }
 }
 
-void RequestHandler::consumePartitions(
-    std::pair<std::string, bool> MessageAndEOF, int &EOFPartitionCounter,
-    FlatbuffersTranslator &FlatBuffers) {
-  if (!MessageAndEOF.first.empty() &&
-      MessageAndEOF.first != "HiddenSecretMessageFromLovingNeutron") {
-    FlatBuffers.getFileID(&MessageAndEOF.first);
+void RequestHandler::consumePartitions(KafkaMessageMetadataStruct &MessageData,
+                                       int &EOFPartitionCounter,
+                                       FlatbuffersTranslator &FlatBuffers) {
+  if (!MessageData.PayloadToReturn.empty() &&
+      MessageData.PayloadToReturn != "HiddenSecretMessageFromLovingNeutron") {
```

Comment:
We have some messages on Kafka in plain JSON (not FlatBuffer serialised) so it would be better to have a flag in `MessageData` to record if the message is text, rather than relying on a magic string. Handling JSON messages should be done as a separate issue, but you could at least use this flag in `ConnectKafkaFake` for now.